controllers/operatorpolicy_controller.go

```diff
@@ -60,166 +70,211 @@ var _ reconcile.Reconciler = &OperatorPolicyReconciler{}
-// create an OperatorGroup in the ns as the Subscription, set the compliance status, and log the policy.
-//
-// In the future, more reconciliation logic will be added. For reference:
-// https://github.com/JustinKuli/ocm-enhancements/blob/89-operator-policy/enhancements/sig-policy/89-operator-policy-kind/README.md
-func (r *OperatorPolicyReconciler) handleSinglePolicy(
+// createPolicyResources encapsulates the logic for creating resources specified within an operator policy.
+// This should normally happen when the policy is initially applied to the cluster. Creates an OperatorGroup
+// if specified, otherwise defaults to using an allnamespaces OperatorGroup.
+func (r *OperatorPolicyReconciler) createPolicyResources(
 	policy *policyv1beta1.OperatorPolicy,
-) error {
-	OpLog.Info("Handling OperatorPolicy", "policy", policy.Name)
+	watcher depclient.ObjectIdentifier,
+) (ctrl.Result, error) {
+	// Retrieve og from cache
+	ogNamespace := policy.Spec.Subscription.Namespace
+	if policy.Spec.OperatorGroup != nil {
+		ogNamespace = policy.Spec.OperatorGroup.Namespace
+	}
 
-	subscriptionSpec := new(operatorv1alpha1.Subscription)
-	err := r.Get(context.TODO(),
-		types.NamespacedName{Namespace: subscriptionSpec.Namespace, Name: subscriptionSpec.Name},
-		subscriptionSpec)
-	exists := !errors.IsNotFound(err)
-	shouldExist := strings.EqualFold(string(policy.Spec.ComplianceType), string(policyv1.MustHave))
+	ogList, err := r.DynamicWatcher.ListFromCache(
+		operatorgroupgvk,
+		ogNamespace, nil)
+	if err != nil {
+		panic(err)
+	}
 
-	// Object does not exist but it should exist, create object
-	if !exists && shouldExist {
+	//  Create og, then trigger reconcile
+	if len(ogList) == 0 {
 		if strings.EqualFold(string(policy.Spec.RemediationAction), string(policyv1.Enforce)) {
-			OpLog.Info("creating kind " + subscriptionSpec.Kind + " in ns " + subscriptionSpec.Namespace)
-			subscriptionSpec := buildSubscription(policy, subscriptionSpec)
-			err = r.Create(context.TODO(), subscriptionSpec)
+			ogSpec := buildOperatorGroup(policy)
 
+			err = r.Create(context.TODO(), ogSpec)
 			if err != nil {
+				OpLog.Error(err, "Error while creating OperatorGroup")
 				r.setCompliance(policy, policyv1.NonCompliant)
-				OpLog.Error(err, "Could not handle missing musthave object")
 
-				return err
+				err = r.DynamicWatcher.EndQueryBatch(watcher)
+				if err != nil {
+					panic(err)
+				}
+
+				return reconcile.Result{}, err
 			}
 
-			// Currently creates an OperatorGroup for every Subscription
-			// in the same ns, and defaults to targeting all ns.
-			// Future implementations will enable targeting ns based on
-			// installModes supported by the CSV. Also, only one OperatorGroup
-			// should exist in each ns
-			operatorGroup := buildOperatorGroup(policy)
-			err = r.Create(context.TODO(), operatorGroup)
+			// Created successfully, requeue result
+			r.setCompliance(policy, policyv1.NonCompliant)
```

Comment:
Why is this non compliant if it was successful?

Reply:
At this point the subscription has not been created yet, so the `NonCompliant` is like a placeholder until the subscription is created. My logic is that the policy should be `NonCompliant` until every object in the spec is created.